chainguard-dev/edu
# Choosing an Image for your Compiled Programs

## Type of change
Add new "cheatsheet" to the Academy, detailing some key use-case differences between various images suitable for compiled programs

### What should this PR do?
resolves https://github.com/chainguard-dev/internal/issues/3798

### Why are we making this change?
Customers have raised concerns regarding differences between these images. For example, some have not understood why they are paying for the `chainguard-base` image instead of using `wolfi-base`. Others have questioned why `musl` is not supported. This article addresses these concerns and more, detailing a comparison between other similar images such as `gcc-glibc`, `glibc-dynamic`, `static`...

### What are the acceptance criteria? 
Some things to note:
- It was suggested by Lewis and I that a link to this cheatsheet is placed in the image overviews (on the registry) for all of the images mentioned in this cheatsheet. That way, customers or curious developers can be linked out to the doc instead of manually comparing // not knowing the doc exists. However, I am not sure who to raise this to or how that can be orchestrated. I figure it would happen after we merge this PR.

- I placed this under the "working with images" section of the Academy, but we can move this around if someone has a better idea.

### How should this PR be tested?
- Please preview the article using the deploy preview link, click the links, make sure everything seems to connect where it should :)
- Technical edits are much appreciated. I pieced this together using what information I could find... since there is not a singular source of truth, I tried my best to gather multiple things together. If there's anything I got wrong, could clarify, add, or do differently (especially considering the customer experience) then it would be much, much appreciated!
